nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,112 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+
+  toConf = { isAttr ? false }@args: v:
+    if builtins.isFloat v then toString v
+    else if isList v then "[ " + concatMapStringsSep ", " (toConf {}) v + " ]"
+    else if isAttr && isString v then toString v
+    else generators.toPretty {} v;
+
+  clightConf = {
+    inherit (config.location) latitude longitude;
+    gamma_temp = with cfg.temperature; [ day night ];
+  } // cfg.settings;
+
+  clightConfFile = pkgs.writeText "clight.conf"
+    (concatStringsSep "\n" (mapAttrsToList
+      (name: value: "${toConf { isAttr = true; } name} = ${toConf {} value};")
```

Comment:
I'd just do `"${toString name} = ${toConf value}"` and get rid of the `isAttr` thing.